cpp/src/copying/scatter.cu

```diff
@@ -393,7 +393,7 @@ std::unique_ptr<column> boolean_mask_scatter(column_view const& input,
 
   // The scatter map is actually a table with only one column, which is scatter map.
   auto scatter_map =
-    detail::apply_boolean_mask(table_view{{indices->view()}}, boolean_mask, stream);
+    detail::apply_boolean_mask(table_view{{indices->view()}}, boolean_mask, stream, mr);
```

Comment:
```suggestion
    detail::apply_boolean_mask(table_view{{indices->view()}}, boolean_mask, stream, rmm::mr::get_current_device_resource());
```
This appears to be a temporary and so should not use the passed in `mr`

Reply:
Ah thanks I wasn't careful enough in checking on these. Will fix.